crates/driver/src/main.rs

```diff
@@ -383,12 +383,13 @@ async fn build_auction_converter(
             .unwrap(),
         );
 
-        Some(ZeroExLiquidity {
-            api: zeroex_api,
-            zeroex: contracts::IZeroEx::deployed(&common.web3).await.unwrap(),
-            base_tokens: base_tokens.clone(),
-            gpv2: common.settlement_contract.clone(),
-        })
+        Some(ZeroExLiquidity::new(
+            common.web3.clone(),
```

Comment:
Should we pass around a `Arc<dyn AllowanceManaging>` instead of creating a new one in each sub-component?

Reply:
Good point! 👍

Reply:
I will do that in a follow up PR.